source/Ship.h

```diff
@@ -448,6 +456,8 @@ class Ship : public Body, public std::enable_shared_from_this<Ship> {
 	// Characteristics of this particular ship:
 	std::string name;
 	bool canBeCarried = false;
+	// The bay this ship docks too, if any.
+	Bay *reservedBay = nullptr;
```

Comment:
I'm still not happy by the direct pointer from one ship to anothers ship bay; all the other code around ships referring other ships is using weak_pointers which suggest that all other code is always assuming that the other ship could disappear any Engine::Step() (which is probably a valid assumption).

If the bays are in a vector, then it should be possible to refer to the bay by using the index-number in the vector. (Such an index number would be similar to a weak pointer in that it is possible to verify if the bay-index is valid before using the bay-index.)

The child/parent relation is relatively strictly defined, so I expect that it is possible to make this code work with a direct pointer, but if we really go down that path, then we need a lot more test-coverage before I consider this ready for merging.

Reply:
I think it's pretty safe, since `reservedBay` is really only used when we know that the escort has a parent and the bay is also part of the parent. But I'll try to find another solution; index sound goods thought, thanks.